Improve folder layout consistency

pjcollins opened the pull request.

rachelkang reviewed (changes requested):
Could you also delete docs/assets as those aren't being used anymore?
Also do we still need .editorconfig?

rachelkang reviewed (approved):
ignore my last review. lgtm!

pjcollins commented:
> Could you also delete docs/assets as those aren't being used anymore? Also do we still need .editorconfig?

I removed the old screenshots and moved native-library-interop-binding-overview.png into this folder to clean up the root, the .editorconfig file can stick around as well
